src/rantanplan/core.py

```diff
@@ -681,21 +743,27 @@ def _get_scansion(text, rhyme_analysis=False, rhythm_format="pattern",
         if rhythmical_lengths is not None:
             structure_length = rhythmical_lengths
         else:
+            # Handle repetitive stanzas
             line_structure = line.get("structure", None)
-            structure_length = STRUCTURES_LENGTH.get(line_structure, None)
-        if structure_length is not None:
+            structure_length, repetitive_structure = STRUCTURES_LENGTH.get(
+                line_structure, [[], False])
+            if structure_length and repetitive_structure:
+                repetitions = int(len(lines) / len(structure_length))
+                structure_length = structure_length * repetitions
+        if structure_length:
```

Comment:
Repetitive and repeating mean different things I believe.
```suggestion
            structure_length, repeating_structure = STRUCTURES_LENGTH.get(
                line_structure, [[], False])
            if structure_length and repeating_structure:
                repetitions = int(len(lines) / len(structure_length))
                structure_length = structure_length * repetitions
        if structure_length:
```